tests/std/tests/P0896R4_ranges_alg_minmax/test.cpp

```diff
@@ -334,6 +336,16 @@ constexpr void mm_constexpr_tests() {
         ProxyRef::no>>();
 }
 
+void test_for_issue1893() {
+    string val         = "meow";
+    string low         = "m";
+    string high        = "n";
+    const auto clamped = ranges::clamp(
+        ref(val), ref(low), ref(high), [](auto x, auto y) { return x < y; },
+        [](const auto& x) -> decltype(auto) { return x.get(); });
+    assert(val == "meow");

```

Comment:
This is not checking whether we actually only use 3 projections.

Can you add some state to the projection?